prov/rxd/src/rxd_av.c

```diff
@@ -84,6 +84,7 @@ static int rxd_av_set_addrlen(struct rxd_av *av, const void *addr)
 	if (ret != 1) {
 		FI_WARN(&rxd_prov, FI_LOG_AV, "addr insert failed: %d (%s)\n",
 			-ret, fi_strerror(-ret));
+		ret = -FI_EINVAL;
```

Comment:
Why not pass ret through?

Reply:
The functions rxd_av_set_addrlen and rxd_av_insert_dg_addr (comment above) both deal with fi_av_insert (which would return 1 on success) and other functions that return error codes. So if we pass ret through on failure (ret = 0) then we can't catch the error.